plugins/outputs/stomp/stomp.go

```diff
@@ -0,0 +1,124 @@
+package stomp
+
+import (
+	"crypto/tls"
+	"fmt"
+	"net"
+	"time"
+
+	"github.com/go-stomp/stomp"
+
+	"github.com/influxdata/telegraf"
+	commontls "github.com/influxdata/telegraf/plugins/common/tls"
+	"github.com/influxdata/telegraf/plugins/outputs"
+	"github.com/influxdata/telegraf/plugins/serializers"
+)
+
+type STOMP struct {
+	Host      string          `toml:"host"`
+	Username  string          `toml:"username"`
+	Password  string          `toml:"password"`
+	QueueName string          `toml:"queueName"`
+	Log       telegraf.Logger `toml:"-"`
+
+	HeartBeatSend time.Duration `toml:"heartBeatSend"`
+	HeartBeatRec  time.Duration `toml:"heartBeatRec"`
+
+	commontls.ClientConfig
+
+	conn  net.Conn
+	stomp *stomp.Conn
+
+	serialize serializers.Serializer
+}
+
+func (q *STOMP) Connect() error {
+	tlsConfig, err := q.ClientConfig.TLSConfig()
+	if err != nil {
+		return err
+	}
+	if tlsConfig != nil {
+		q.conn, err = tls.Dial("tcp", q.Host, tlsConfig)
+		if err != nil {
+			return err
+		}
+	} else {
+		q.conn, err = net.Dial("tcp", q.Host)
+		if err != nil {
+			return err
+		}
+	}
+
+	q.stomp, err = stomp.Connect(q.conn, stomp.ConnOpt.HeartBeat(q.HeartBeatSend, q.HeartBeatRec), stomp.ConnOpt.Login(q.Username, q.Password))
+	if err != nil {
+		return err
+	}
+	q.Log.Debug("STOMP Connected...")
+	return nil
+}
+
+func (q *STOMP) SetSerializer(serializer serializers.Serializer) {
+	q.serialize = serializer
+}
+
+func (q *STOMP) Write(metrics []telegraf.Metric) error {
+	for _, metric := range metrics {
+		values, err := q.serialize.Serialize(metric)
+		if err != nil {
+			q.Log.Errorf("Serializing metric %v failed: %s", metric, err)
+			continue
+		}
+		err = q.stomp.Send(q.QueueName, "text/plain", values, nil)
+		if err != nil {
+			return fmt.Errorf("sending metric failed: %s", err)
+		}
+	}
+	return nil
+}
+
+func (q *STOMP) Close() error {
+	return q.stomp.Disconnect()
+}
+
+func (q *STOMP) SampleConfig() string {
+	return `
+  ## Host of Active Mq broker
+  host = "localhost:61613"
+  ## Queue name for producer messages
+  # queueName = "telegraf"
+  ## Optional credentials
+  # username = ""
+  # password = ""
+
+  ## Optional HeatBeats Configuration
+  #heartBeatSend = 0
+  #heartBeatRec = 0
```

Comment:
Maybe use
```suggestion
  # heartBeatSend = "0s"
  # heartBeatRec = "0s"
```
to make clear that you can specify a duration here.